### ISSUE TITLE
PullRequestListener.rescopedEvent calls canMerge for declined pull requests

### ISSUE THREAD

**Issue by bturner** [Eernie/bitbucket-webhooks-plugin#56]
`PullRequestListener.rescopedEvent` calls `PullRequestService.canMerge` without first checking the state of the pull request on the event. When a pull request is rescoped and declined "close" together, it triggers errors like this:

```
2017-08-02 09:02:55,847 ERROR [AtlassianEvent::thread-9]  c.a.s.i.e.AsyncBatchingInvokersTransformer There was an exception thrown trying to dispatch event 'com.atlassian.bitbucket.event.pull.PullRequestRescopedEvent[source=com.atlassian.stash.internal.pull.DefaultPullRequestService@2a0b98d]' for the invoker 'SingleParameterMethodListenerInvoker{method=public void nl.topicus.bitbucket.api.PullRequestListener.rescopedEvent(com.atlassian.bitbucket.event.pull.PullRequestRescopedEvent) throws java.io.IOException, listener=nl.topicus.bitbucket.api.PullRequestListener@2fc0a10}'
java.lang.RuntimeException: This pull request has been declined and must be reopened before it can be merged.. Listener: nl.topicus.bitbucket.api.PullRequestListener event: com.atlassian.bitbucket.event.pull.PullRequestRescopedEvent
	at com.atlassian.event.internal.SingleParameterMethodListenerInvoker.invoke(SingleParameterMethodListenerInvoker.java:55)
	at com.atlassian.stash.internal.event.AsyncBatchingInvokersTransformer$AsyncInvokerBatch.invoke(AsyncBatchingInvokersTransformer.java:109)
	at com.atlassian.event.internal.AsynchronousAbleEventDispatcher$1$1.run(AsynchronousAbleEventDispatcher.java:38)
	at com.atlassian.sal.core.executor.ThreadLocalDelegateRunnable.run(ThreadLocalDelegateRunnable.java:34)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.lang.Thread.run(Thread.java:748)
	... 1 frame trimmed
Caused by: com.atlassian.bitbucket.pull.IllegalPullRequestStateException: This pull request has been declined and must be reopened before it can be merged.
	at com.atlassian.stash.internal.pull.DefaultPullRequestService.checkIsNotDeclined(DefaultPullRequestService.java:1223)
	at com.atlassian.stash.internal.pull.DefaultPullRequestService.checkIsOpen(DefaultPullRequestService.java:1235)
	at com.atlassian.stash.internal.pull.DefaultPullRequestService.canMerge(DefaultPullRequestService.java:244)
	at com.atlassian.plugin.util.ContextClassLoaderSettingInvocationHandler.invoke(ContextClassLoaderSettingInvocationHandler.java:26)
	at org.eclipse.gemini.blueprint.service.importer.support.internal.aop.ServiceInvoker.doInvoke(ServiceInvoker.java:56)
	at org.eclipse.gemini.blueprint.service.importer.support.internal.aop.ServiceInvoker.invoke(ServiceInvoker.java:60)
	at org.eclipse.gemini.blueprint.service.util.internal.aop.ServiceTCCLInterceptor.invokeUnprivileged(ServiceTCCLInterceptor.java:70)
	at org.eclipse.gemini.blueprint.service.util.internal.aop.ServiceTCCLInterceptor.invoke(ServiceTCCLInterceptor.java:53)
	at org.eclipse.gemini.blueprint.service.importer.support.LocalBundleContextAdvice.invoke(LocalBundleContextAdvice.java:57)
	at nl.topicus.bitbucket.api.PullRequestListener.rescopedEvent(PullRequestListener.java:94)
	at com.atlassian.event.internal.SingleParameterMethodListenerInvoker.invoke(SingleParameterMethodListenerInvoker.java:40)
	... 6 common frames omitted
```

Before calling `canMerge`, `rescopedEvent` should first check `isOpen()` on the event's `PullRequest`. If that's `true`, then it can call `canMerge`. If not, then it shouldn't. It should _also_ explicitly handle `IllegalPullRequestStateException` because, even with the check, there's always going to be room for race conditions between pull request updates and event processing.

**Comment by casz** [comment]
@bturner would you mind taking a look at #55

**Comment by bturner** [comment]
#57 also includes a fix for this issue.

**Comment by casz** [comment]
Fixed in 1.5.0 👍
